// protoc --python_out=. message.proto

syntax = "proto2";

package wol;

message Auth {
    optional string name = 1 [default = "Etaoin"];
}

message AuthResponse {
    optional bool success = 1 [default = false];
    optional string error = 2 [default = "Uknown error"];
    optional int64 your_id = 3 [default = -1];
}

message PleaseSendAllIDs {}

message PleaseSendState {
    optional int64 target_id = 1;
}

message CreateObj {}

message NewObjectCreated {
    optional int64 object_id = 1;
}

message UserAction {
    optional int32 type = 1;
    optional int64 target_id = 2;
}


message ObjectStateUpdate{
    optional int64 object_id = 1;

    message FieldUpdate {
        optional string field_name = 2;
        repeated string string_value = 3;
        repeated int64 int_value = 4;
        repeated float float_value = 5;
    }

    repeated FieldUpdate field = 2;
}

message RemoveObject {
    optional int64 target_object_id = 1;
    optional int64 remover_id = 2;
}

message Message {
    oneof content {
        Auth auth = 1;
        AuthResponse auth_response = 2;
        PleaseSendAllIDs please_send_all_ids = 3;
        PleaseSendState please_send_state = 4;
        CreateObj create_obj = 5;
        UserAction user_actions = 6;
        NewObjectCreated new_object_create= 7;
        ObjectStateUpdate object_state_update = 8;
    }
}